source/includes/_protocol_schemas.mdown

```diff
@@ -0,0 +1,215 @@
+|**description** | A full textual description of the listing. Linebreaks are currently ignored, though this will change. |
+|**price** | For currencies, we support `ETH`, with future plans to support any ERC20 token |
+|**unitsTotal** | The number of units available for sale. Currently limited to `1` |
+
+### Optional Fields
+
+| | |
+|----|-----|
+|expiry | A date that this listing should no longer show in search results. This is suggestion and search engines may ignore it.|
+|media | A list of media elements to be shown with the listing. We currently only support **images** on the **ipfs://** protocol.|
+|commission | This is the amount the seller is willing to pay the DApp that sells the listing user after the sale is complete. The currency must be `OGN`. This money comes out of the deposit amount the seller puts into the listing. Note that the deposit amount may very well be too low to actualy pay the commission - this must be checked for before trusting the commission.|
+
+
+## Offer Schema
+
+An Offer is sent by a buyer to make a purchase from a listing.
+
+```javascript
+{
+    schemaId: "http://schema.originprotocol.com/offer_v1.0.0",
+    listingType: "unit",
+    unitsPurchased: 1,
+    totalPrice: {amount: "48.7", currency: "ETH"},
+    commission: {amount: "500", currency: "OGN"}
+}
+```
+
+### Required Fields:
+
+| | |
+|----|-----|
+|**schemaId** | Must be set to `http://schema.originprotocol.com/offer_v1.0.0` |
+|**listingType** | Must be set to `unit` |
+|**unitsPurchased** | The number of units to purchase. Currently must be `1`. |
+|**totalPrice** | The price the buyer is willing to pay. For currencies, we support `ETH`, with future plans to support any ERC20 token. |
+|**commission**| **UNSAFE FIELD - this data duplicates, and may not match, the actual blockchain contract amount.** The amount the buyer's DApp will receive for the sale. Must be equal to, or less than, the listing's commission amount. The currency must be `OGN`. The amount may be zero if there is no listing commission set.|
+
+
+## Dispute Schema
+
+A dispute can be started by either party. In v1.0.0, this schema contains no useful required fields, and is placeholder for future expansion. It is intended that the parties and the arbitrator will communicate over Origin Messenger after the dispute process is started.
+
+```javascript
+{
+    schemaId: "http://schema.originprotocol.com/dispute_v1.0.0",
+}
+```
+
+### Required Fields:
+
+| | |
+|----|-----|
+|**schemaId** | Must be set to `http://schema.originprotocol.com/dispute_v1.0.0` |
+
+
+## Listing-Withdraw Schema
+
+In v1.0.0, this schema contains no useful required fields, and is placeholder for future expansion.
+
+```javascript
+{
+    schemaId: "http://schema.originprotocol.com/listing-withdraw_v1.0.0",
+}
+```
+
+### Required Fields:
+
+| | |
+|----|-----|
+|**schemaId** | Must be set to `http://schema.originprotocol.com/listing-withdraw_v1.0.0` |
+
+
+## Notification Schema
```

Comment:
I don't think notifications are stored in IPFS.
Also it does not look like the schema under src/IpsInterface/schema/notification.json is actually used at all. I can't find any code referencing it to do any sort of validation. I should probably delete it ?

Maybe we should document this under the "models" which are objects returned as output of Origin-js methods ?
The notification model is [here](https://github.com/OriginProtocol/origin-js/blob/master/src/models/notification.js).

Reply:
I've removed this from the docs.